Bug: Cross-module function resolution fails when using `using` directive

## Description

When using the `using` directive to access functions from a local module (not stdlib), the typechecker incorrectly reports E4002 (undefined function) errors. This happens because the typechecker doesn't have knowledge of functions defined in other modules when checking function bodies.

## Steps to Reproduce

1. Create a multi-file project with:
   - `models.ez` - defines a function like `get_priority_name()`
   - `main.ez` - imports models and uses `using models`
2. Call `get_priority_name()` directly in main.ez (without the `models.` prefix)
3. Run typechecker

## Expected Behavior

The function call should be recognized as valid since `using models` brings the function into scope.

## Actual Behavior

```
error[E4002]: undefined function 'get_priority_name'
  --> main.ez:87:33
   |
87 |     temp priority_name string = get_priority_name(task.priority)
   |                                 ^ function not defined
```

## Affected Tests

- `integration-tests/pass/multi-file/basic/`
- `integration-tests/pass/multi-file/foreach-namespaced-collection/`
- `integration-tests/pass/multi-file/imports/`

## Root Cause

The typechecker currently:
1. Recognizes stdlib functions via `using` (println, sqrt, etc.) through hardcoded lists
2. Does NOT recognize user-defined functions from local modules via `using`

The fix requires the typechecker to:
1. Collect all function signatures from all modules in a multi-file project BEFORE checking function bodies
2. When checking a function call with `using`, look up the function in all "used" modules' function tables

## Workaround

Use the fully qualified module prefix:
```ez
// Instead of:
get_priority_name(task.priority)

// Use:
models.get_priority_name(task.priority)
```

## Related

- Discovered while fixing #663 (undefined variable/function detection)
- The stdlib `using` case works because those functions are hardcoded in `checkDirectStdlibCall()`

--- a/pkg/typechecker/typechecker.go
+++ b/pkg/typechecker/typechecker.go
@@ -3585,6 +3585,40 @@ func (tc *TypeChecker) isKnownIdentifier(name string) bool {
 			}
 		}
 	}
+	// Check if it's a function from a user module accessible via 'using' (#671)
+	for moduleName := range tc.fileUsingModules {
+		if funcs, ok := tc.moduleFunctions[moduleName]; ok {
+			if _, exists := funcs[name]; exists {
+				return true
+			}
+		}
+	}
+	if tc.currentScope != nil {
+		for moduleName := range tc.currentScope.usingModules {
+			if funcs, ok := tc.moduleFunctions[moduleName]; ok {
+				if _, exists := funcs[name]; exists {
+					return true
+				}
+			}
+		}
+	}
+	// Check if it's a type from a user module accessible via 'using' (#671)
+	for moduleName := range tc.fileUsingModules {
+		if types, ok := tc.moduleTypes[moduleName]; ok {
+			if _, exists := types[name]; exists {
+				return true
+			}
+		}
+	}
+	if tc.currentScope != nil {
+		for moduleName := range tc.currentScope.usingModules {
+			if types, ok := tc.moduleTypes[moduleName]; ok {
+				if _, exists := types[name]; exists {
+					return true
+				}
+			}
+		}
+	}
 	return false
 }
 
@@ -4494,7 +4528,52 @@ func (tc *TypeChecker) checkDirectStdlibCall(funcName string, call *ast.CallExpr
 		}
 	}
 
+	// Check user-defined module functions accessible via 'using' (#671)
+	for moduleName := range tc.fileUsingModules {
+		if funcs, ok := tc.moduleFunctions[moduleName]; ok {
+			if sig, exists := funcs[funcName]; exists {
+				// Validate argument count
+				tc.validateCallArguments(funcName, call, sig, line, column)
+				return true
+			}
+		}
+	}
+	if tc.currentScope != nil {
+		for moduleName := range tc.currentScope.usingModules {
+			if funcs, ok := tc.moduleFunctions[moduleName]; ok {
+				if sig, exists := funcs[funcName]; exists {
+					tc.validateCallArguments(funcName, call, sig, line, column)
+					return true
+				}
+			}
+		}
+	}
+
 	return false
+}
+
+// validateCallArguments validates argument count for a function call (#671)
+func (tc *TypeChecker) validateCallArguments(funcName string, call *ast.CallExpression, sig *FunctionSignature, line, column int) {
+	// Calculate minimum required arguments (parameters without defaults)
+	minRequired := 0
+	for _, param := range sig.Parameters {
+		if !param.HasDefault {
+			minRequired++
+		}
+	}
+
+	// Check argument count
+	if len(call.Arguments) < minRequired || len(call.Arguments) > len(sig.Parameters) {
+		var msg string
+		if minRequired == len(sig.Parameters) {
+			msg = fmt.Sprintf("wrong number of arguments to '%s': expected %d, got %d",
+				funcName, len(sig.Parameters), len(call.Arguments))
+		} else {
+			msg = fmt.Sprintf("wrong number of arguments to '%s': expected %d to %d, got %d",
+				funcName, minRequired, len(sig.Parameters), len(call.Arguments))
+		}
+		tc.addError(errors.E5008, msg, line, column)
+	}
 }
 
 // isMathFunction checks if a function name exists in the math module

